knowledge-base/kb-diagram-custom-connection.md

```diff
@@ -0,0 +1,228 @@
+---
+title: Create Custom RadDiagramConnection
+page_title: Custom RadDiagramConnection 
+description: Currently the RadDiagramConnection and Cap elements are represented by one Path object. To achieve further customization, extend the base implementation of the RadDiagramConnection element.
+type: how-to
+slug: kb-diagram-custom-connection-cap
+position: 0
+tags: diagram, connection, cap
+ticketid: 1542456
+res_type: kb
+category: knowledge-base
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product Version</td>
+			<td>2021.1.325</td>
+		</tr>
+		<tr>
+			<td>Product</td>
+			<td>Diagram for WPF</td>
+		</tr>
+	</tbody>
+</table>
+
+## Description
+
+With the current implementation of the __RadDiagram__ control applying properties to a __RadDiagramConnection__ element will also apply them to the Connection Cap. This is because the __ConnectionLine__ and __ConnectionCap__ elements are visualized by a single native __Path__ control. To further customize the separate elements, the default implementation of the __RadDiagramConnection__ control, would need to be extended.
+
+## Solution
+
+To extend the base implementation of the __RadDiagramConnection__ element, extract its default control template, and add two additional __Path__ controls.
```

Comment:
Add link to the Editing Control Templates article.

Add a complete and customized ControlTemplate, so it is easier to copy/paste it. You can use the NoXaml approach in order to be easier for customization and also mention this in the article.

Add note saying which theme the extracted template is based on.

Add a picture showing the end result in the beginning. This way, the reader will see if this article matches its requirement if it is unsure.